zerver/models/custom_profile_fields.py

```diff
@@ -168,6 +169,8 @@ def as_dict(self) -> ProfileDataElementBase:
         }
         if self.display_in_profile_summary:
             data_as_dict["display_in_profile_summary"] = True
+        if self.required:
+            data_as_dict["required"] = True
```

Comment:
This is probably worth an "#api design" conversation -- I'm not entirely sure why we made `display_in_profile_summary` an optional field, but I think it's possible the reasoning for that doesn't apply to this field, and it should just be always present. (I guess this explains why you had that type in the JS).

Reply:
Fixed in the recent commit.